### PR TITLE
refactor: change nexus path as agred on with Infra

### PR DESCRIPTION
**Pull request by kate-shine**
Related issue: https://github.com/famedly/infra-meta/issues/929#issuecomment-1674724750

### CODE REVIEW

File: .github/workflows/docker-backend.yml [hunk 1]
```diff
@@ -9,19 +9,13 @@ on:
...
 
       path:
         required: false
         type: string
         default: "."
-        
-      repository:
-        required: false
-        type: string
-        default: ${{ github.repository }}
```
Comment (FlixCoder) [inline .github/workflows/docker-backend.yml:24]: Wait, why did you remove this, when it is used in the `IMAGE_NAME` still?

File: .github/workflows/docker-backend.yml [hunk 2]
```diff
@@ -9,19 +9,13 @@ on:
       # Name of service. Empty for root service, needs to be unique for services from subfolders. 
       # Will be used as a last part of container path and matched with version tags
       name:
-        required: false
+        required: true
         type: string
-        default: ""
 
       path:
         required: false
```
Comment (FlixCoder) [inline .github/workflows/docker-backend.yml:16]: Due to the step that validates the occurrence of inputs, this is required, too, while it should not. This needs more adjustments I think ^^

**Review by FlixCoder — CHANGES_REQUESTED** [review]
Are we using a different domain by default too, i.e. the private docker?

File: .github/workflows/docker-backend.yml (hunk 1 above)
Comment (kate-shine) [reply .github/workflows/docker-backend.yml:24]: Oh? I was searching that variable ;_;

Comment (kate-shine) [reply .github/workflows/docker-backend.yml:24]: Wait, it doesn't, IMAGE_NAME only uses registry ^_^

File: .github/workflows/docker-backend.yml (hunk 2 above)
Comment (kate-shine) [reply .github/workflows/docker-backend.yml:16]: It's more that we don't need that check now :)

File: .github/workflows/docker-backend.yml (hunk 1 above)
Comment (FlixCoder) [reply .github/workflows/docker-backend.yml:24]: oh oops

File: .github/workflows/docker-backend.yml [hunk 3]
```diff
@@ -6,22 +6,16 @@ on:
         type: string
         default: docker.nexus.famedly.de
 
-      # Name of service. Empty for root service, needs to be unique for services from subfolders. 
+      # Name of service.
       # Will be used as a last part of container path and matched with version tags
```
Comment (FlixCoder) [inline .github/workflows/docker-backend.yml:10]: if path is specified, right?

Comment (kate-shine) [reply .github/workflows/docker-backend.yml:10]: I don't understand?

Comment (FlixCoder) [reply .github/workflows/docker-backend.yml:10]: the comment should probably be extended:
```suggestion
      # Will be used as a last part of container path and matched with version tags if path is specified
```

**Comment by kate-shine** [comment]
> I would still set the default registry to the private one, what do you think?

Apparently it's `docker` and `docker-oss` now? (See linked issue)

**Review by FlixCoder — APPROVED** [review]

File: .github/workflows/docker-backend.yml [hunk 4]
```diff
@@ -6,22 +6,17 @@ on:
         type: string
         default: docker.nexus.famedly.de
```
Comment (emgrav) [inline .github/workflows/docker-backend.yml:7]: I think it could be good to add a comment here to be extra clear, even if this is documented in notion

```suggestion
        # docker.nexus.famedly.de is the private container registry
        default: docker.nexus.famedly.de
```

File: .github/workflows/docker-backend.yml [hunk 5]
```diff
@@ -34,17 +29,10 @@ jobs:
   docker:
     runs-on: ubuntu-latest-16core
     env:
-      # conditionally adds '/' if inputs.name isn't empty (https://github.com/orgs/community/discussions/25725)
-      IMAGE_NAME: "${{ inputs.registry }}/${{ inputs.repository }}${{ inputs.name && '/' || '' }}${{ inputs.name }}"
+      IMAGE_NAME: "${{ inputs.registry }}/backend/${{ inputs.name }}"
```
Comment (emgrav) [inline .github/workflows/docker-backend.yml:32]: I don't think `IMAGE_NAME` is a good name for this variable anymore, as it ultimately represents the full path. Maybe `IMAGE_PATH`?

**Review by emgrav — CHANGES_REQUESTED** [review]
I just have two very small remarks, but otherwise this looks great.

**Review by emgrav — APPROVED** [review]
